### PR TITLE
typeshed: remove typing sins from re.Match.groups

### PR DESCRIPTION
**Pull request by KotlinIsland**
amon gus

### CODE REVIEW

File: mypy/util.py [hunk 1]
```diff
@@ -824,3 +824,8 @@ def plural_s(s: int | Sized) -> str:
         return "s"
     else:
         return ""
+
+
+def safe(o: T | None) -> T:
```
Comment (DetachHead) [inline mypy/util.py:829]: should this take a `message` argument as well

File: mypy/typeshed/stdlib/re.pyi [hunk 2]
```diff
@@ -74,19 +74,19 @@ class Match(Generic[AnyStr]):
     @overload
     def group(self, __group: Literal[0] = 0) -> AnyStr: ...
     @overload
-    def group(self, __group: str | int) -> AnyStr | Any: ...
+    def group(self, __group: str | int) -> AnyStr | None: ...
```
Comment (DetachHead) [inline mypy/typeshed/stdlib/re.pyi:77]: are these changes also being added to basedtypeshed

Comment (KotlinIsland) [reply mypy/typeshed/stdlib/re.pyi:77]: Thanks, currently the implementation of basedtypeshed is a but ruff. I'll fix this up there.

File: mypy/util.py (hunk 1 above)
Comment (KotlinIsland) [reply mypy/util.py]: Hows that look?

Comment (DetachHead) [reply mypy/util.py]:
```py
def safe(o: T | None, msg: str) -> T:
    ...

print(safe(foo, "foo was None"))
```

also what `o` means? should it be renamed to something more descriptive like `value`?

**Review by DetachHead — APPROVED** [review]

**Pull request merged**